### DISCUSSION TITLE
OpenXLA and PyTorch 2.0

### DISCUSSION THREAD

**Discussion by kevint324** [openxla/community#30] [Project Discussions]
Hi Folks,

Is there a roadmap about how to integrate XLA into PyTorch 2.0 with the new released compile infrastructure.

Thanks

**Comment by shauheen** [comment]
Thank you for your question. PyTorch and XLA have had an ongoing integration for a few years which you can find at https://github.com/pytorch/xla

PyTorch/XLA is part of the PyTorch foundation and is being maintained by multiple partners who use XLA devices. PyTorch/XLA integration of TorchDynamo [is ongoing](https://github.com/pytorch/xla/pulls?q=is%3Apr+label%3Adynamo+) and will be 2.0 at the same time as PyTorch. As XLA transitions to OpenXLA, PyTorch/XLA will also transition to use OpenXLA and produce [StableHLO](https://github.com/openxla/stablehlo).

If you have any questions or feature requests please feel free to [share them](https://github.com/pytorch/xla/issues/new/choose) on the PyTorch/XLA repo.

**Comment by kevint324** [comment]
Hi Shauheen,

Thanks for the reply. Really appreciate. A few more question

![image](https://user-images.githubusercontent.com/8800468/207810479-69d1a41d-6fc1-4e1d-b658-c071eff2b900.png)

According to the block diagram does XLA fits in the "Others"  green block at codegen backends layer?
And is there any plan for Pytorch XLA GPU support?  [Here ](https://pytorch.org/xla/master/#gpu) it says "Coming soon in a future release!".

Thank you.

Kevin

**Reply by shauheen** [reply]
Thank you Kevin for following up. The diagram may not be exact (at least not in short term) but generally you are correct XLA is another backend. I recommend [this post](https://dev-discuss.pytorch.org/t/torchdynamo-update-10-integrating-with-pytorch-xla-for-inference-and-training/935) for more detailed information on the status of the integration with TorchDynamo. As mentioned in the post PyTorch/XLA already works on GPU, however as you see in the documentation we have not been providing official support yet.

**Reply by kevint324** [reply]
Thank you Shauheen. These information are helpful.
